### PR TITLE
Decimal Refactor

### PR DESCRIPTION
**Pull request by zengyan1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6611 

## What this PR does / why we need it:
New Decimal64 and Decimal128

### CODE REVIEW

**Review by reusee — APPROVED** [review]

**Comment by sukki37** [comment]
The code coverage of some files is below 75%. Please add more unit tests or BVT cases.
image

**Review by ouyuanning — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Comment by sukki37** [comment]
Please add some ut or bvt cases to improve code coverage, or at least try to improve the coverage of your own code files, thanks.
image

File: pkg/vm/engine/tae/compute/compare.go [hunk 1]
```diff
@@ -73,9 +73,9 @@ func CompareGeneric(a, b any, t types.Type) int64 {
 	case types.T_uint64:
 		return CompareOrdered[uint64](a, b)
 	case types.T_decimal64:
-		return types.CompareDecimal64Decimal64Aligned(a.(types.Decimal64), b.(types.Decimal64))
+		return int64(a.(types.Decimal64).Compare(b.(types.Decimal64)))
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/compute/compare.go:76]: return types.CompareDecimal64(a.(types.Decimal64), b.(types.Decimal64))

File: pkg/vm/engine/tae/compute/compare.go [hunk 2]
```diff
@@ -73,9 +73,9 @@ func CompareGeneric(a, b any, t types.Type) int64 {
...
 	case types.T_decimal128:
-		return types.CompareDecimal128Decimal128Aligned(a.(types.Decimal128), b.(types.Decimal128))
+		return int64(a.(types.Decimal128).Compare(b.(types.Decimal128)))
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/compute/compare.go:78]: return types.CompareDecimal128(a.(types.Decimal128), b.(types.Decimal128))

File: pkg/vm/engine/tae/containers/vecwindow.go [hunk 3]
```diff
@@ -89,13 +89,13 @@ func (win *vectorWindow[T]) Equals(o Vector) bool {
 		} else if _, ok := any(v).(types.Decimal64); ok {
 			d := win.Get(i).(types.Decimal64)
 			od := win.Get(i).(types.Decimal64)
-			if d.Ne(od) {
+			if d != od {
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/containers/vecwindow.go:92]: return d == od

File: pkg/vm/engine/tae/containers/vecwindow.go [hunk 4]
```diff
@@ -89,13 +89,13 @@ func (win *vectorWindow[T]) Equals(o Vector) bool {
...
 				return false
 			}
 		} else if _, ok := any(v).(types.Decimal128); ok {
 			d := win.Get(i).(types.Decimal128)
 			od := win.Get(i).(types.Decimal128)
-			if d.Ne(od) {
+			if d != od {
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/containers/vecwindow.go:98]: return d == od

**Review by XuPeng-SH — APPROVED** [review]

**Review by LeftHandCold — APPROVED** [review]

File: pkg/container/types/decimal_test.go [hunk 5]
```diff
@@ -15,120 +15,160 @@
...
+		panic("DecimalAddSub wrong")
+	}
+}
 
-	d128, err = Decimal128_FromString(longstr)
-	require.True(t, err == nil)
-	dd, err = d128.ToDecimal64(33, 15)
-	require.True(t, d64.Eq(dd))
-	require.True(t, err == nil)
+func TestDecimalMulDiv(t *testing.T) {
```
Comment (aunjgr) [inline pkg/container/types/decimal_test.go:50]: This case fails. Please fix and recommit.

**Review by aunjgr — CHANGES_REQUESTED** [review]

**Review by m-schen — APPROVED** [review]
colexec and function lgtm.

**Review by aunjgr — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

**Pull request merged**